monkey/monkey_island/cc/services/authentication_service/flask_resources/agent_otp_login.py

```diff
@@ -26,27 +32,61 @@ def post(self):
         """
         Gets the one-time password from the request,
         and returns an authentication token and a refresh token
+        for a particular Agent
 
-        :return: Authentication token in the response body
+        :return: Authentication token and refresh token in the response body
         """
-
         try:
-            cred_dict = json.loads(request.data)
-            otp = cred_dict.get("otp", "")
-            if self._validate_otp(otp):
-                refresh_token = "refreshtoken"
+            agent_id, otp = self._get_request_arguments(request.json)
+        except Exception as err:
+            return make_response(str(err), HTTPStatus.BAD_REQUEST)
```

Comment:
This may be easier to read, but it's wrong. If this raises an exception other than Type, Value, or Key, that would be an internal server error, not a BAD REQUEST error. I'm going to drop these commits to get this back to working.

Reply:
I've resolved this by adding an exception type internal to the endpoint.